packages/react/transform/crates/swc_plugin_snapshot/lib.rs

```diff
@@ -804,8 +812,18 @@ where
 
       let is_list = jsx_is_list(n);
       let is_children_full_dynamic = is_list || jsx_is_children_full_dynamic(n);
+      let is_single_static_text = jsx_is_single_static_text(n);
+      let is_ge_3_1 = is_engine_version_ge(&self.cfg.engine_version, "3.1");
 
-      if !is_children_full_dynamic {
+      if is_single_static_text && is_ge_3_1 {
+        if let JSXElementChild::JSXText(text) = &n.children[0] {
+          self.static_stmts.push(RefCell::new(quote!(
+            r#"__SetAttribute($element, "text", $value)"# as Stmt,
+            element: Ident = el.clone(),
+            value: Expr = Expr::Lit(Lit::Str(jsx_text_to_str(&text.value).into())),
+          )));
+        }
```

Comment:
I prefer not to couple this logic to the full_dynamic check, as adding new rules later might complicate the `is_children_full_dynamic` check. I think we could have a dedicated text plugin that runs before the snapshot plugin to perform this compilation:

before:

```js
<view>Some single text</view>
```

after:

```js
<view text="Some single text" />
```

Reply:
- Yes, we have list plugin which will change the jsx structure from `<list-item defer/>` to `<DeferredListItem />`, see: https://github.com/lynx-family/lynx-stack/blob/42961787956aef7322bffcfc955bac5fd82b43f3/packages/react/transform/crates/swc_plugin_list/lib.rs#L72
- No, swc will handle the sourcemap and it will work fine.

I think we should add a new `text_plugin` here (just before the `snapshot_plugin`): https://github.com/lynx-family/lynx-stack/blob/42961787956aef7322bffcfc955bac5fd82b43f3/packages/react/transform/src/lib.rs#L555 In  this way, any fututer features related to text can be implemented in it.